src/main/groovy/org/pih/warehouse/inventory/CycleCountItemDto.groovy

```diff
@@ -22,6 +22,8 @@ class CycleCountItemDto {
 
     Integer quantityCounted
 
+    Integer discrepancy
+
     DiscrepancyReasonCode discrepancyReasonCode
```

Comment:
There might be a time when we need to distinguish between reason code by activity (cycle count, allocation, picking, adjustments, etc) but when that happens we're probably going to build an extensible data model around it. 

For now, we need to remove the DiscrepancyReasonCode enum and replace it here with ReasonCode enum.

![image](https://github.com/user-attachments/assets/bbbae141-c118-48aa-9746-ca2b01227daf)

Source: https://openboxes.slack.com/archives/C06JCHWR807/p1738168383197519

Reply:
@kchelstowski @EWaterman

Reply:
it's currently an open question. Manon met with some of the Boston crew (Ali and Ginny) and they were unclear about the codes they want to display here. Likely it is going to be a subset of ReasonCode values but we've haven't gotten a conclusive answer yet.

But you're right we won't need a custom enum here.

For now we should probably use ReasonCode.listInventoryAdjustmentReasonCodes() and then have the frontend call ReasonCodeApiController.list() to fetch the list of possible values.

But we can do that in a separate PR once we know a bit more